catalogs/catalog-messaging-kafka/src/main/java/com/datastrato/gravitino/catalog/kafka/KafkaCatalogOperations.java

```diff
@@ -92,39 +123,237 @@ public void initialize(Map<String, String> config, CatalogInfo info) throws Runt
+
+      } else if (change instanceof TopicChange.SetProperty) {
+        TopicChange.SetProperty setProperty = (TopicChange.SetProperty) change;
+        // alter partition count
+        if (PARTITION_COUNT.equals(setProperty.getProperty())) {
+          int targetPartitionCount = Integer.parseInt(setProperty.getValue());
+          if (targetPartitionCount == newPartitionCount) {
+            continue;
+          } else if (targetPartitionCount < newPartitionCount) {
+            throw new IllegalArgumentException(
+                "Cannot reduce partition count from "
+                    + newPartitionCount
+                    + " to "
+                    + targetPartitionCount);
+          } else {
+            newPartitionCount = targetPartitionCount;
+            alteredProperties.put(PARTITION_COUNT, setProperty.getValue());
+            continue;
+          }
+        }
+
+        // alter other properties
+        alteredProperties.put(setProperty.getProperty(), setProperty.getValue());
+        alterConfigOps.add(
+            new AlterConfigOp(
+                new ConfigEntry(setProperty.getProperty(), setProperty.getValue()),
+                AlterConfigOp.OpType.SET));
+
+      } else if (change instanceof TopicChange.RemoveProperty) {
+        TopicChange.RemoveProperty removeProperty = (TopicChange.RemoveProperty) change;
+        Preconditions.checkArgument(
+            !PARTITION_COUNT.equals(removeProperty.getProperty()), "Cannot remove partition count");
+        alteredProperties.remove(removeProperty.getProperty());
+        alterConfigOps.add(
+            new AlterConfigOp(
+                new ConfigEntry(removeProperty.getProperty(), null), AlterConfigOp.OpType.DELETE));
+
+      } else {
+        throw new IllegalArgumentException("Unsupported topic change: " + change);
+      }
+    }
+
+    // increase partition count
+    if (newPartitionCount != oldPartitionCount) {
+      try {
+        adminClient
+            .createPartitions(
+                Collections.singletonMap(ident.name(), NewPartitions.increaseTo(newPartitionCount)))
+            .all()
+            .get();
+      } catch (Exception e) {
+        throw new RuntimeException(
+            "Failed to increase partition count for topic " + ident.name(), e);
+      }
+    }
+
+    // alter topic properties
+    if (!alterConfigOps.isEmpty()) {
+      ConfigResource topicResource = new ConfigResource(ConfigResource.Type.TOPIC, ident.name());
+      try {
+        adminClient
+            .incrementalAlterConfigs(Collections.singletonMap(topicResource, alterConfigOps))
+            .all()
+            .get();
+      } catch (UnknownTopicOrPartitionException e) {
+        throw new NoSuchTopicException(e, "Topic %s does not exist", ident);
+      } catch (Exception e) {
+        throw new RuntimeException("Failed to alter topic properties for topic " + ident.name(), e);
+      }
+    }
```

Comment:
Can you please split the method to split the partition change related code another method?

Also do we need to check if the replication factor can be changed, if not should you add some check here?

Reply:
> Can you please split the method to split the partition change related code another method?

done

> Also do we need to check if the replication factor can be changed, if not should you add some check here?

It's unnecessary because the replication factor is immutable, and properties metadata will ensure that